tests/unit/core/TestAddonManager.js

```diff
@@ -125,70 +124,68 @@ describe(__filename, () => {
   });
 
   describe('install()', () => {
-    it('should call mozAddonManager.createInstall() with url', () =>
-      addonManager
-        .install(fakeInstallUrl, fakeCallback, {
-          _mozAddonManager: fakeMozAddonManager,
-          src: 'home',
-        })
-        .then(() => {
-          expect(
-            fakeMozAddonManager.createInstall.calledWith({
-              url: `${fakeInstallUrl}?src=home`,
-            }),
-          ).toBeTruthy();
-        }));
-
-    it('should call installObj.addEventListener to setup events', () =>
-      addonManager
-        .install(fakeInstallUrl, fakeCallback, {
-          _mozAddonManager: fakeMozAddonManager,
-          src: 'home',
-        })
-        .then(() => {
-          // It registers an extra onInstallFailed and onInstallEnded listener.
-          expect(fakeInstallObj.addEventListener.callCount).toEqual(
-            INSTALL_EVENT_LIST.length + 2,
-          );
-        }));
-
-    it('should call installObj.install()', () =>
-      addonManager
-        .install(fakeInstallUrl, fakeCallback, {
-          _mozAddonManager: fakeMozAddonManager,
-          src: 'home',
-        })
-        .then(() => {
-          expect(fakeInstallObj.install.called).toBeTruthy();
-        }));
+    it('should call mozAddonManager.createInstall() with url', async () => {
+      await addonManager.install(fakeInstallUrl, fakeCallback, {
+        _mozAddonManager: fakeMozAddonManager,
+        src: 'home',
+      });
 
-    it('rejects if the install fails', () => {
+      sinon.assert.calledWith(fakeMozAddonManager.createInstall, {
+        url: `${fakeInstallUrl}?src=home`,
+      });
+    });
+
+    it('should call installObj.addEventListener to setup events', async () => {
+      await addonManager.install(fakeInstallUrl, fakeCallback, {
+        _mozAddonManager: fakeMozAddonManager,
+        src: 'home',
+      });
+
+      // It registers an extra onInstallFailed and onInstallEnded listener.
+      sinon.assert.callCount(
+        fakeInstallObj.addEventListener,
+        INSTALL_EVENT_LIST.length + 2,
+      );
+    });
+
+    it('should call installObj.install()', async () => {
+      await addonManager.install(fakeInstallUrl, fakeCallback, {
+        _mozAddonManager: fakeMozAddonManager,
+        src: 'home',
+      });
+
+      sinon.assert.called(fakeInstallObj.install);
+    });
+
+    it('rejects if the install fails', async (done) => {
       fakeInstallObj.install = sinon.spy(function install() {
         this.onInstallFailedListener();
       });
-      return addonManager
-        .install(fakeInstallUrl, fakeCallback, {
+
+      try {
+        await addonManager.install(fakeInstallUrl, fakeCallback, {
           _mozAddonManager: fakeMozAddonManager,
           src: 'home',
-        })
-        .then(unexpectedSuccess, () =>
-          expect(fakeInstallObj.install.called).toBeTruthy(),
-        );
+        });
+
+        done.fail(new Error('this should not be called'));
```

Comment:
This pattern is a bit clunky to me. I think it's more succinct in promise form (as the test was before) and it's less prone to errors such as forgetting to call `done()`.

Reply:
reverted